src/bp/ui-admin/src/Pages/Server/Monitoring.jsx

```diff
@@ -0,0 +1,335 @@
+import ChartTooltip from '../Components/Monitoring/ChartTooltip'
+import LoadingSection from '../Components/LoadingSection'
+
+import { groupEntriesByTime, mergeEntriesByTime, calculateOverviewForHost } from 'common/monitoring'
+import { fetchStats, refreshStats } from '../../reducers/monitoring'
+
+const timeFrameOptions = [
+  { value: '1m', label: '1 minute' },
+  { value: '5m', label: '5 minutes' },
+  { value: '10m', label: '10 minutes' },
+  { value: '15m', label: '15 minutes' },
+  { value: '30m', label: '30 minutes' },
+  { value: '1h', label: '1 hour' },
+  { value: '2h', label: '2 hours' },
+  { value: '5h', label: '5 hours' }
+]
+
+const resolutionOptions = [
+  { value: '10s', label: '10 seconds' },
+  { value: '30s', label: '30 seconds' },
+  { value: '1m', label: '1 minutes' },
+  { value: '5m', label: '5 minutes' },
+  { value: '15m', label: '15 minutes' },
+  { value: '30m', label: '30 minutes' },
+  { value: '1h', label: '1 hour' }
+]
+
+const tickSize = { fontSize: 11 }
+
+class Monitoring extends Component {
+  state = {
+    intervalId: null,
+    timeFrame: null,
+    resolution: null,
+    rawStats: null,
+    preparedStats: null,
+    timeFrameOptions
+  }
+
+  componentDidMount() {
+    this.setState({ timeFrame: timeFrameOptions[3], resolution: _.head(resolutionOptions) }, this.queryData)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.rawStats !== this.props.rawStats) {
+      this.prepareForDisplay()
+    }
+  }
+
+  async queryData() {
+    if (!this.state.timeFrame) {
+      return
+    }
+
+    try {
+      const fromTime = moment()
+        .subtract(ms(this.state.timeFrame.value))
+        .toDate()
+        .getTime()
+
+      const toTime = moment()
+        .toDate()
+        .getTime()
+
+      this.props.fetchStats(fromTime, toTime)
+    } catch (error) {
+      this.setState({ error: error.message })
+    }
+  }
+
+  prepareForDisplay() {
+    if (!this.props.rawStats || !this.props.rawStats.length) {
+      return
+    }
+
+    const lastUniqueEntries = _.sortBy(_.uniqBy(_.orderBy(this.props.rawStats, ['ts'], ['desc']), 'host'), 'host')
+    const uniqueHosts = _.map(lastUniqueEntries, 'host')
+    this.setState({ uniqueHosts, lastUniqueEntries })
+
+    // Group results by interval, then calculates total and average
+    const grouped = groupEntriesByTime(this.props.rawStats, this.state.resolution.value)
+    const merged = mergeEntriesByTime(grouped, uniqueHosts)
+
+    this.setState({ preparedStats: merged })
+  }
+
+  handleTimeFrameChanged = timeFrame => this.setState({ timeFrame }, this.queryData)
+  handleResolutionChanged = resolution => this.setState({ resolution }, this.queryData)
+
+  handleAutoRefreshChanged = event => {
+    const autoRefresh = event.target.checked
+
+    this.setState({ autoRefresh }, () => {
+      if (autoRefresh) {
+        const intervalId = setInterval(() => {
+          this.props.refreshStats()
+        }, 5000)
```

Comment:
It would be nice if Auto refresh interval being the same as `collectionInterval` configured in config. Even if they're not exactly at the same tick. It would prevent to fetch for nothing if the interval is higher than 5s

Reply:
Yep, but collectionInterval can be started at any time depending on when the node starts, so if there's 10 nodes, they could each post results each seconds